src/mpi/pt2pt/bsendutil.c

```diff
@@ -558,6 +639,7 @@ static int MPIR_Bsend_progress(struct MPII_BsendBuffer_user *user)
         MPIR_Request *req = active->request;
         if (MPIR_Request_is_complete(req)) {
             MPIR_Bsend_free_segment(user, active);
+            /* FIXME: how can req be persistent? Is this a left-over */
```

Comment:
are the requests here child or user requests?

Reply:
Real request from `MPID_Isend` as far as I can tell - https://github.com/pmodels/mpich/blob/340f174e58b514ba6d62f5dc2224befd0d8f3146/src/mpi/pt2pt/bsendutil.c#L485-L486.

Reply:
I think this is likely a leftover from a refactor. Let me remove the persistent check and see if we break any tests.